pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -184,6 +210,168 @@ func (r *MachineSyncReconciler) reconcileCAPIMachinetoMAPIMachine(ctx context.Co
+	}
+
+	if result, err := r.createOrUpdateCAPIMachine(ctx, mapiMachine, capiMachine, newCAPIMachine); err != nil {
+		return result, fmt.Errorf("unable to ensure CAPI machine: %w", err)
+	}
+
+	return ctrl.Result{}, r.updateSynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionTrue,
+		consts.ReasonResourceSynchronized, messageSuccessfullySynchronized, &mapiMachine.Generation)
+}
+
+// convertMAPIToCAPIMachine converts a MAPI Machine to a CAPI Machine, selecting the correct converter based on the platform.
+func (r *MachineSyncReconciler) convertMAPIToCAPIMachine(mapiMachine *machinev1beta1.Machine) (*capiv1beta1.Machine, client.Object, []string, error) {
+	switch r.Platform {
+	case configv1.AWSPlatformType:
+		return mapi2capi.FromAWSMachineAndInfra(mapiMachine, r.Infra).ToMachineAndInfrastructureMachine() //nolint:wrapcheck
+	case configv1.PowerVSPlatformType:
+		return mapi2capi.FromPowerVSMachineAndInfra(mapiMachine, r.Infra).ToMachineAndInfrastructureMachine() //nolint:wrapcheck
+	default:
+		return nil, nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, r.Platform)
+	}
+}
+
+// createOrUpdateCAPIInfraMachine creates a CAPI infra machine from a MAPI machine, or updates if it exists and it is out of date.
+func (r *MachineSyncReconciler) createOrUpdateCAPIInfraMachine(ctx context.Context, mapiMachine *machinev1beta1.Machine, infraMachine client.Object, newCAPIInfraMachine client.Object) (ctrl.Result, error) { //nolint:unparam
+	logger := log.FromContext(ctx)
+
+	if infraMachine == nil {
+		if err := r.Create(ctx, newCAPIInfraMachine); err != nil {
+			logger.Error(err, "Failed to create CAPI infra machine")
+			createErr := fmt.Errorf("failed to create CAPI infra machine: %w", err)
+
+			if condErr := r.updateSynchronizedConditionWithPatch(ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToCreateCAPIInfraMachine, createErr.Error(), nil); condErr != nil {
+				return ctrl.Result{}, utilerrors.NewAggregate([]error{createErr, condErr})
+			}
+
+			return ctrl.Result{}, createErr
+		}
+
+		logger.Info("Successfully created CAPI infra machine")
+
+		return ctrl.Result{}, nil
+	}
+
+	isEqualCAPIInfraMachine, err := capiInfraMachineIsEqual(r.Platform, infraMachine, newCAPIInfraMachine)
+	if err != nil {
+		logger.Error(err, "Failed to check CAPI infra machine diff")
+		updateErr := fmt.Errorf("failed to check CAPI infra machine diff: %w", err)
+
+		if condErr := r.updateSynchronizedConditionWithPatch(
+			ctx, mapiMachine, corev1.ConditionFalse, reasonFailedToUpdateCAPIInfraMachine, updateErr.Error(), nil); condErr != nil {
+			return ctrl.Result{}, utilerrors.NewAggregate([]error{updateErr, condErr})
+		}
+
+		return ctrl.Result{}, updateErr
+	}
+
+	if isEqualCAPIInfraMachine {
+		logger.Info("No changes detected in CAPI infra machine")
+		return ctrl.Result{}, nil
+	}
+
+	logger.Info("Updating CAPI infra machine")
+
+	if err := r.Update(ctx, newCAPIInfraMachine); err != nil {
```

Comment:
The only updates we really expect are metadata right?

Reply:
Let's talk about that here: https://github.com/openshift/cluster-capi-operator/pull/258#discussion_r1963236079